File: src/main/resources/output/IFC4X3_csharp/Entities/IfcRoot/IfcObjectDefinition/IfcObject/IfcProduct/IfcElement/IfcBuiltElement/IfcDoor/IfcDoor.cs
public class IfcDoor : IfcBuiltElement
{
    public IfcPositiveLengthMeasure OverallHeight { get; set; }
    public IfcPositiveLengthMeasure OverallWidth { get; set; }
    public IfcDoorTypeEnum PredefinedType { get; set; }
    public IfcDoorTypeOperationEnum OperationType { get; set; }
    public IfcLabel UserDefinedOperationType { get; set; }

    // === WHERE CLAUSES ===
    // CorrectPredefinedType : NOT(EXISTS(PredefinedType)) OR (PredefinedType <> IfcDoorTypeEnum.USERDEFINED) OR ((PredefinedType = IfcDoorTypeEnum.USERDEFINED) AND EXISTS (SELF\IfcObject.ObjectType))
    // CorrectTypeAssigned : (SIZEOF(IsTypedBy) = 0) OR ('IFC4X3_DEV_73740fe4.IFCDOORTYPE' IN TYPEOF(SELF\IfcObject.IsTypedBy[1].RelatingType))
}

File: src/main/resources/output/IFC4X3_python/Entities/IfcRoot/IfcObjectDefinition/IfcObject/IfcProduct/IfcElement/IfcBuiltElement/IfcDoor/IfcDoor.py
class IfcDoor(IfcBuiltElement):
    def __init__(self):
        self.OverallHeight: IfcPositiveLengthMeasure = None
        self.OverallWidth: IfcPositiveLengthMeasure = None
        self.PredefinedType: IfcDoorTypeEnum = None
        self.OperationType: IfcDoorTypeOperationEnum = None
        self.UserDefinedOperationType: IfcLabel = None

    # === WHERE CLAUSES ===
    # CorrectPredefinedType : NOT(EXISTS(PredefinedType)) OR (PredefinedType <> IfcDoorTypeEnum.USERDEFINED) OR ((PredefinedType = IfcDoorTypeEnum.USERDEFINED) AND EXISTS (SELF\IfcObject.ObjectType))
    # CorrectTypeAssigned : (SIZEOF(IsTypedBy) = 0) OR ('IFC4X3_DEV_73740fe4.IFCDOORTYPE' IN TYPEOF(SELF\IfcObject.IsTypedBy[1].RelatingType))
